### PR TITLE
Update sql prompt on db change

### PR DESCRIPTION
**Pull request by sjanusz-r7**
This PR is a UX improvement. It adds the correct handling for changing of databases in the context of a MySQL or MSSQL session. The prompt now gets updated with an up-to-date value.
This PR also aligns the SQL clients to use the `current_database` method.

### Before
```ruby
MySQL @ 127.0.0.1:3306 () > query show databases
Response
========

    #  Database
    -  --------
    0  information_schema
    1  my_docker_db
    2  my_new_db
    3  mysql
    4  performance_schema
    5  sys

MySQL @ 127.0.0.1:3306 () > query use mysql
[-] Error running command query: NoMethodError undefined method `entries' for nil:NilClass
MySQL @ 127.0.0.1:3306 () > query use garbage
[-] Query has failed. Reasons: Unknown database 'garbage'
```

### After
```ruby
MySQL @ 127.0.0.1:3306 () > query show databases
Response
========

    #  Database
    -  --------
    0  information_schema
    1  my_docker_db
    2  my_new_db
    3  mysql
    4  performance_schema
    5  sys

MySQL @ 127.0.0.1:3306 () > query use mysql
[*] Query executed successfully
MySQL @ 127.0.0.1:3306 (mysql) > query use sys
[*] Query executed successfully
MySQL @ 127.0.0.1:3306 (sys) > query use garbage
[-] Query has failed. Reasons: Unknown database 'garbage'
```

### Before
```ruby
MSSQL @ 192.168.112.3:1433 (master) > query select name from master.dbo.sysdatabases
Response
========

    #  name
    -  ----
    0  master
    1  tempdb
    2  model
    3  msdb

MSSQL @ 192.168.112.3:1433 (master) > query use msdb
[-] Error running command query: NoMethodError undefined method `map' for nil:NilClass
MSSQL @ 192.168.112.3:1433 (msdb) > query use master
[-] Error running command query: NoMethodError undefined method `map' for nil:NilClass
MSSQL @ 192.168.112.3:1433 (master) > query use garbage
[-] Query has failed. Reasons: SQL Server Error #911 (State:1 Severity:16): Database 'garbage' does not exist. Make sure that the name is entered correctly.
```

## After
```ruby
MSSQL @ 192.168.112.3:1433 (master) > query select name from master.dbo.sysdatabases
Response
========

    #  name
    -  ----
    0  master
    1  tempdb
    2  model
    3  msdb

MSSQL @ 192.168.112.3:1433 (master) > query use msdb
[*] Query executed successfully
MSSQL @ 192.168.112.3:1433 (msdb) > query use master
[*] Query executed successfully
MSSQL @ 192.168.112.3:1433 (master) > query use garbage
[-] Query has failed. Reasons: SQL Server Error #911 (State:1 Severity:16): Database 'garbage' does not exist. Make sure that the name is entered correctly.
```

## Bogus & Correct Queries
The following ensures we get the correct output for good and bogus queries:

```ruby
MSSQL @ 192.168.112.3:1433 (master) > query use hello
[-] Query has failed. Reasons: SQL Server Error #911 (State:1 Severity:16): Database 'hello' does not exist. Make sure that the name is entered correctly.
MSSQL @ 192.168.112.3:1433 (master) > query use msdb
[*] Query executed successfully
MSSQL @ 192.168.112.3:1433 (msdb) > query select @@version
Response
========

    #  NULL
    -  ----
    0  Microsoft SQL Server 2022 (RTM) - 16.0.1000.6 (X64)
        Oct  8 2022 05:58:25
        Copyright (C) 2022 Microsoft Corporation
        Express Edition (64-bit) on Windows Server 2022 Standard 10.0 <X64> (Build 20348: ) (Hypervisor)

...

MySQL @ 127.0.0.1:3306 (sys) > query use hello
[-] Query has failed. Reasons: Unknown database 'hello'
MySQL @ 127.0.0.1:3306 (sys) > query show databases
Response
========

    #  Database
    -  --------
    0  information_schema
    1  my_docker_db
    2  my_new_db
    3  mysql
    4  performance_schema
    5  sys

...

PostgreSQL @ 127.0.0.1:5432 (template1) > query select version
[-] Query has failed. Reasons: ERROR    VERROR  C42703  Mcolumn "version" does not exist        P8      Fparse_relation.c       L3713   RerrorMissingColumn
PostgreSQL @ 127.0.0.1:5432 (template1) > query select version()
[*] SELECT 1

Response
========

    #  version
    -  -------
    0  PostgreSQL 16.1 (Debian 16.1-1.pgdg120+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 12.2.0-14) 12.2.0, 64-bit
```

## Verification

- [ ] Get a PostgreSQL, MSSQL and MySQL session
- [ ] **Verify** you can change the database by typing in `query use ....`
- [ ] Ensure typing in correct queries results in the expected output with no errors
- [ ] Ensure typing in bogus queries results in errors being output

### CODE REVIEW

File: lib/rex/post/sql/ui/console/command_dispatcher/client.rb [hunk 1]
```diff
@@ -137,11 +137,31 @@ def cmd_query(*args)
                 end
 
                 result = run_query(args.join(' '))
+                case result[:status]
+                when :success
+                  # When changing a database in MySQL, we get a nil result back.
+                  # When changing a database in MSSQL, we get a result, but it doesn't contain colnames or rows.
+                  normalised_result = normalise_sql_result(result[:result])
```
Comment (sjanusz-r7) [inline lib/rex/post/sql/ui/console/command_dispatcher/client.rb:144]: Potentially:
```suggestion
                  normalised_result = normalise_sql_result(result[:result]) unless result[:result].nil?
```
Then investigate what effect that will have on the rest of this method, or deal with `nil` result in the `normalise_sql_result` method.

**Comment by cgranleese-r7** [comment]
Everything worked as expected 👍 

# MySQL
## Before
image

## After
image

# MSSQL
## Before
image

## After
image

# PostgreSQL
image

File: lib/rex/post/sql/ui/console/command_dispatcher/client.rb [hunk 2]
```diff
@@ -137,11 +137,31 @@ def cmd_query(*args)
...
+
+                  # MSSQL returns :success, even if the query failed due to wrong syntax.
+                  if normalised_result[:errors].any?
+                    print_error "Query has failed. Reasons: #{normalised_result[:errors].join(', ')}"
+                    return
+                  end
 
-                normalised_result = result[:status] == :success ? normalise_sql_result(result[:result]) : result[:result]
-                formatted_result = format_result(normalised_result)
+                  if normalised_result[:rows].nil? || normalised_result[:columns].nil? # Should this be AND?
```
Comment (cgranleese-r7) [inline lib/rex/post/sql/ui/console/command_dispatcher/client.rb:152]: Did you happen to figure out if this should have been an AND?

Comment (cgranleese-r7) [reply lib/rex/post/sql/ui/console/command_dispatcher/client.rb:152]: Calling it out as I seen the comment 😄

File: lib/rex/post/sql/ui/console/command_dispatcher/client.rb (hunk 1 above)
Comment (adfoster-r7) [reply lib/rex/post/sql/ui/console/command_dispatcher/client.rb:144]: Anything actionable here?

File: lib/rex/post/sql/ui/console/command_dispatcher/client.rb (hunk 2 above)
Comment (sjanusz-r7) [reply lib/rex/post/sql/ui/console/command_dispatcher/client.rb:152]: Will keep this as an OR in an upcoming commit. Rex Table throws an exception if nil values are provided, so we need an OR here 👍

File: lib/rex/post/sql/ui/console/command_dispatcher/client.rb (hunk 1 above)
Comment (sjanusz-r7) [reply lib/rex/post/sql/ui/console/command_dispatcher/client.rb:153]: Changed in latest commits to explicitly handle `nil` result 👍

**Comment by cgranleese-r7** [comment]
Tested again after the recent changes and everything seems to be working as expected 👍 
## MySQL
image

## MSSQL
image

**Comment by cgranleese-r7** [comment]
## Release Notes
This PR improves the UX by correctly handling databases changes by updating the prompt  to now get the appropriate database value. in the context of a MySQL or MSSQL session.

**Pull request merged**